static/app/views/alerts/rules/metric/edit.tsx

```diff
@@ -24,91 +29,83 @@ type Props = {
   userTeamIds: string[];
 } & RouteComponentProps<RouteParams, {}>;
 
-type State = {
-  actions: Map<string, any>;
-  rule: MetricRule; // This is temp
-} & DeprecatedAsyncView['state'];
-
-class MetricRulesEdit extends DeprecatedAsyncView<Props, State> {
-  getDefaultState() {
-    return {
-      ...super.getDefaultState(),
-      actions: new Map(),
-    };
-  }
-
-  getTitle(): string {
-    const {organization, project} = this.props;
-    const {rule} = this.state;
-    const ruleName = rule?.name;
-
-    return routeTitleGen(
-      ruleName ? t('Alert - %s', ruleName) : '',
-      organization.slug,
-      false,
-      project?.slug
-    );
-  }
-
-  getEndpoints(): ReturnType<DeprecatedAsyncView['getEndpoints']> {
-    const {organization} = this.props;
-    const {ruleId} = this.props.params;
-
-    return [['rule', `/organizations/${organization.slug}/alert-rules/${ruleId}/`]];
-  }
-
-  onRequestSuccess({stateKey, data}) {
-    if (stateKey === 'rule' && data.name) {
-      this.props.onChangeTitle(data.name);
+export function MetricRulesEdit({
+  organization,
+  params,
+  project,
+  userTeamIds,
+  onChangeTitle,
+  ...props
+}: Props) {
+  const navigate = useNavigate();
+
+  const {
+    isLoading,
+    isError,
+    data: rule,
+    error,
+  } = useApiQuery<MetricRule>(
+    [`/organizations/${organization.slug}/alert-rules/${params.ruleId}/`],
+    {
+      staleTime: 0,
+      retry: false,
+      onSuccess: data => {
+        onChangeTitle(data[0].name ?? '');
```

Comment:
Array access by index is not typesafe and can result in undefined.
```suggestion
        onChangeTitle(data[0]?.name ?? '');
```